Makefile

```diff
@@ -76,33 +76,37 @@ deepcopy-gen:
 
 image-nmi:
 	cp bin/$(PROJECT_NAME)/$(NMI_BINARY_NAME) images/nmi
-	docker build -t $(NMI_IMAGE_NAME):$(NMI_VERSION) images/nmi
+	docker build -t $(NMI_IMAGE_NAME) images/nmi
 
 image-mic:
 	cp bin/$(PROJECT_NAME)/$(MIC_BINARY_NAME) images/mic
-	docker build -t $(MIC_IMAGE_NAME):$(MIC_VERSION) images/mic
+	docker build -t $(MIC_IMAGE_NAME) images/mic
 
 image-demo:
 	cp bin/$(PROJECT_NAME)/$(DEMO_BINARY_NAME) images/demo
-	docker build -t $(DEMO_IMAGE_NAME):$(DEMO_VERSION) images/demo
+	docker build -t $(DEMO_IMAGE_NAME) images/demo
 
 image-identity-validator:
 	cp bin/$(PROJECT_NAME)/$(IDENTITY_VALIDATOR_BINARY_NAME) images/identityvalidator
-	docker build -t $(IDENTITY_VALIDATOR_IMAGE_NAME):$(IDENTITY_VALIDATOR_VERSION) images/identityvalidator
+	docker build -t $(IDENTITY_VALIDATOR_IMAGE_NAME) images/identityvalidator
 
 image:image-nmi image-mic image-demo image-identity-validator
 
 push-nmi:
-	docker push $(NMI_IMAGE_NAME):$(NMI_VERSION)
+	docker pull $(NMI_IMAGE_NAME); if [[ $$? -eq 0 ]]; then echo "$(NMI_IMAGE_NAME) already exists" && exit 1; fi
+	docker push $(NMI_IMAGE_NAME)
 
 push-mic:
-	docker push $(MIC_IMAGE_NAME):$(MIC_VERSION)
+	docker pull $(MIC_IMAGE_NAME); if [[ $$? -eq 0 ]]; then echo "$(MIC_IMAGE_NAME) already exists" && exit 1; fi
+	docker push $(MIC_IMAGE_NAME)
 
 push-demo:
-	docker push $(DEMO_IMAGE_NAME):$(DEMO_VERSION)
+	docker pull $(DEMO_IMAGE_NAME); if [[ $$? -eq 0 ]]; then echo "$(DEMO_IMAGE_NAME) already exists" && exit 1; fi
```

Comment:
Can we please use 'docker images' instead of pull so that we don't have to bring the image to the local system, instead just check if it's present.